lib/paginated.rb

```diff
@@ -0,0 +1,44 @@
+module Paginated
+  def filter_and_paginate(sym)
+    original_method = :"#{sym}_no_pagination"
+    alias_method original_method, sym
+    define_method sym do |page, text, *args|
+      results = send(original_method, *args)
+      filtered = self.class.filter(results, text)
+      self.class.paginate(filtered, page, @config_options)
+    end
+  end
+
+  def filter(results, text)
+    if text.empty?
+      results
+    elsif
+      text = text.downcase
+      results.select { |record| record[:item].values.to_s.downcase.include?(text) }
+    end
+  end
+
+  def paginate(results, page, config_options)
+    if results.length <= config_options[:page_size]
```

Comment:
If `:page_size` is not defined in `config_options` it will be `nil`. So it will try to copare `Int <= nil`.

If we want to ensure `:page_size` is present, use the form `config_options.fetch(:page_size)`. This form allows you to provide a default with a block: `config_options.fetch(:page_size) { 100 }`.

Alternatively, instead of passing a hash of config options, we could use keywords:

```ruby
def paginate(results, page, page_size: 100)
...
end
```

Reply:
After reviewing the whole PR, I guess the above could be reduced to: use `config_options.fetch(:page_size)` or `def paginate(results, page, page_size:)` (note the trailing `:`.

Reply:
`:page_size` will always be defined in `config_options` (see `registers_client.rb`), as will `:cache_duration`.

Reply:
Ok 👍 then I think we can get better errors if we use `#fetch` instead of `#[]`.